core/src/main/java/org/elasticsearch/ingest/core/Processor.java

```diff
@@ -49,7 +54,6 @@
          * Implementations are responsible for removing the used keys, so that after creating a pipeline ingest can
          * verify if all configurations settings have been used.
          */
-        P create(Map<String, Object> config) throws Exception;
-
+        P create(String tag, Map<String, Object> config) throws Exception;
```

Comment:
ah, instead of leaking it as it is now, where it is being handled by the `Pipeline`. yeah. this makes sense. I'll go ahead and do that